Update note on mapping with SAML 2.0

jeremyyeo opened the pull request:
## Description & motivation
SAML 2.0 responses don't send over the name of the group so instead we have to use the ID of the group for mapping.

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

jeremyyeo committed:
Update enterprise-permissions.md

randypitcherii reviewed (approved):
👈 😎

mirnawong1 committed:
Merge branch 'current' into jeremyyeo-patch-5

mirnawong1 commented:
hey @jeremyyeo, thanks for your patience and for opening up this PR. I've approved this and it's good to merge if you'd like!

mirnawong1 committed:
Update enterprise-permissions.md

moved the note above video as it could get missed if users don't scroll further

mirnawong1 commented:
I've gone ahead and merged this for you @jeremyyeo , thanks again for your patience
